lib/livebook_app.ex

```diff
@@ -1,96 +1,103 @@
 if Mix.target() == :app do
-  defmodule LivebookApp do
+  defmodule WxConstants do
     @moduledoc false
 
-    @behaviour :wx_object
+    defmacro wxID_ANY, do: -1
+    defmacro wxID_OPEN, do: 5000
+    defmacro wxID_EXIT, do: 5006
+    defmacro wxID_OSX_HIDE, do: 5250
+    defmacro wxBITMAP_TYPE_PNG, do: 15
+  end
 
-    # https://github.com/erlang/otp/blob/OTP-24.1.2/lib/wx/include/wx.hrl#L1314
-    @wx_id_exit 5006
-    @wx_id_osx_hide 5250
+  defmodule LivebookApp do
+    @moduledoc false
+    @name __MODULE__
 
-    def start_link(_) do
-      {:wx_ref, _, _, pid} = :wx_object.start_link(__MODULE__, [], [])
-      {:ok, pid}
-    end
+    use GenServer
+    import WxConstants
 
-    def child_spec(init_arg) do
-      %{
-        id: __MODULE__,
-        start: {__MODULE__, :start_link, [init_arg]},
-        restart: :transient
-      }
+    def start_link(arg) do
+      GenServer.start_link(__MODULE__, arg, name: @name)
     end
 
-    def windows_connected(url) do
-      url
-      |> String.trim()
-      |> String.trim_leading("\"")
-      |> String.trim_trailing("\"")
-      |> windows_to_wx()
-    end
+    taskbar_icon_path = "static/images/taskbar_icon.png"
```

Comment:
I'm adding a separate icon for the taskbar. (see at the bottom of the diff) The only change at the moment is I removed extra space around the logo so it looks better in the menubar. In the long run we should have an icon with just black on transparent background (no colours) for macOS since that's a convention.